plutus-contract/src/Plutus/Contract/Test/ContractModel/MissingLovelace.hs

```diff
@@ -1,21 +1,20 @@
+{-# LANGUAGE ViewPatterns #-}
 module Plutus.Contract.Test.ContractModel.MissingLovelace
   ( calculateDelta
   ) where
 
-import Data.List as List
+import Data.Set qualified as Set
 import Ledger.Ada qualified as Ada
 import Ledger.Value (Value, noAdaValue)
 import PlutusTx.Prelude qualified as P
 import Wallet.Emulator (Wallet)
 
-calculateDelta :: Value -> Value -> Value -> Wallet -> [(Wallet, [Value])] -> Value
-calculateDelta initialDelta initialValue finalValue w txOutCosts =
+calculateDelta :: Value -> Value -> Value -> Wallet -> [(Wallet, [Ada.Ada])] -> Value
+calculateDelta initialDelta initialValue finalValue w (Set.toList . Set.fromList -> txOutCosts) =
   let
     deltas =
-      let txOutCosts' = case List.uncons $ reverse $ filter ((== w) . fst) txOutCosts of
-            Just ((_, vs), _) -> map Ada.fromValue vs
-            _                 -> []
-          otherWalletsTxOutCosts = concatMap (map Ada.fromValue . snd) $ txOutCosts
+      let txOutCosts' = concatMap snd $ filter ((== w) . fst) txOutCosts
+          otherWalletsTxOutCosts = concatMap snd txOutCosts
```

Comment:
Isn't this `allWalletsTxOutCosts`?

Reply:
Yeah, indeed, thanks for pointing this out. I will refactor and write a comment a bit later.